ui/components/app/smart-transactions/smart-transactions-opt-in-modal.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  TextColor,
+  Display,
+  FlexDirection,
+  BlockSize,
+  AlignItems,
+  TextAlign,
+  JustifyContent,
+  TextVariant,
+  IconColor,
+} from '../../../helpers/constants/design-system';
+import {
+  Modal,
+  ModalOverlay,
+  Text,
+  Box,
+  Button,
+  ButtonVariant,
+  ModalHeader,
+  ModalContent,
+  ButtonLink,
+  ButtonLinkSize,
+  AvatarIcon,
+  IconName,
+  AvatarIconSize,
+} from '../../component-library';
+import { setSmartTransactionsOptInStatus } from '../../../store/actions';
+import { SMART_TRANSACTIONS_LEARN_MORE_URL } from '../../../../shared/constants/smartTransactions';
+
+export type SmartTransactionsOptInModalProps = {
+  isOpen: boolean;
+};
+
+export default function SmartTransactionsOptInModal({
+  isOpen,
+}: SmartTransactionsOptInModalProps) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  const onEnableSmartTransactions = () => {
+    dispatch(setSmartTransactionsOptInStatus(true));
+  };
+  const onNotRightNow = () => {
```

Comment:
Minor, minimal benefit here but best practice is to use `useCallback` for handlers within a component so a single function is created and updated only when needed, rather than creating a fresh function on each render.